### PR TITLE
Dreamview:No close loop function for default routing 

### PR DESCRIPTION
**Pull request by lj0919**

### CODE REVIEW

File: modules/dreamview/frontend/src/store/route_editing_manager.js [hunk 1]
```diff
@@ -139,15 +140,28 @@ export default class RouteEditingManager {
         }
     }
 
-    sendCycleRoutingRequest(defaultRoutingName, cycleNumber) {
-        const points = this.defaultRoutings[defaultRoutingName];
+    sendCycleRoutingRequest(cycleNumber) {
+        const points = this.defaultRoutings[this.currentDefaultRouting];
         if (!isNaN(cycleNumber) || !points) {
-          const success = RENDERER.sendCycleRoutingRequest(defaultRoutingName, points, cycleNumber);
+            const success = RENDERER.sendCycleRoutingRequest
```
Comment (changsh726) [inline modules/dreamview/frontend/src/store/route_editing_manager.js:146]: wrong indent here

File: modules/drivers/lidar/robosense/decoder/decoder_base.hpp [hunk 2]
```diff
@@ -368,11 +368,11 @@ inline double DecoderBase<T_Point>::getLidarTemperature() {
 template <typename T_Point>
 inline void DecoderBase<T_Point>::loadCalibrationFile(
     const std::string& angle_path) {
-  std::cout<<"path! "<<angle_path<<std::endl;
+  std::cout << "path! " << angle_path << std::endl;
   std::string line_str;
   std::ifstream fd_angle(angle_path.c_str(), std::ios::in);
   if (fd_angle.is_open()) {
-      std::cout<<"oepn! "<<std::endl;
+      std::cout << "oepn! " << std::endl;
```
Comment (changsh726) [inline modules/drivers/lidar/robosense/decoder/decoder_base.hpp:375]: fix typo

File: modules/dreamview/frontend/src/renderer/routing_editor.js [hunk 3]
```diff
@@ -135,20 +139,6 @@ export default class RoutingEditor {
...
       }
-
-      checkDefaultRoutingAvailable(start, end, threshold) {
-        if (_.isEmpty(start) || _.isEmpty(end)) {
-          return false;
-        }
-        const distance = Math.sqrt(Math.pow((end.x - start.x), 2) + Math.pow((end.y - start.y), 2));
-        return distance <= threshold;
-      }
 }
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/routing_editor.js:144]: leave a blank line

File: modules/dreamview/frontend/src/components/DefaultRouting/CycleNumberInput.js [hunk 4]
```diff
@@ -1,29 +1,49 @@
...
-    const cycleNumber = parseInt(this.state.cycleNumber, 10);
-    if (isNaN(cycleNumber) || cycleNumber < 1) {
-      alert('please input a valid cycle number');
+    if (this.state.isCycling) {
+      const cycleNumber = parseInt(this.state.cycleNumber, 10);
+      if (isNaN(cycleNumber) || cycleNumber < 1) {
+        alert('please input a valid cycle number');
+      }
+      else if (!routeEditingManager.checkCycleRoutingAvailable()) {
+            alert(`Please set the default routing reasonably,the distance from the start point
```
Comment (changsh726) [inline modules/dreamview/frontend/src/components/DefaultRouting/CycleNumberInput.js:37]: The format seems odd here

File: modules/dreamview/frontend/src/renderer/routing_editor.js [hunk 5]
```diff
@@ -92,16 +92,20 @@ export default class RoutingEditor {
         }
     }
 
-    sendRoutingRequest(carOffsetPosition, carHeading, coordinates) {
-        if (this.routePoints.length === 0) {
+    sendRoutingRequest(carOffsetPosition, carHeading, coordinates, routingPoints) {
+        if (this.routePoints.length === 0 && routingPoints.length === 0) {
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/routing_editor.js:96]: There are two indent styles in this file: 2 spaces and 4 spaces. Please unify into the former style.

Comment (lj0919) [reply modules/dreamview/frontend/src/renderer/routing_editor.js:96]: thx

File: modules/dreamview/frontend/src/renderer/routing_editor.js (hunk 3 above)
Comment (lj0919) [reply modules/dreamview/frontend/src/renderer/routing_editor.js:144]: thx

File: modules/drivers/lidar/robosense/decoder/decoder_base.hpp (hunk 2 above)
Comment (lj0919) [reply modules/drivers/lidar/robosense/decoder/decoder_base.hpp:375]: You little clever ghost！

File: modules/dreamview/frontend/src/store/route_editing_manager.js (hunk 1 above)
Comment (lj0919) [reply modules/dreamview/frontend/src/store/route_editing_manager.js:146]: Ok

**Review by changsh726 — APPROVED** [review]

**Pull request merged**